core/src/main/java/org/jdbi/v3/tweak/ContainerFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.tweak;
+
+import org.jdbi.v3.ContainerBuilder;
+
+/**
+ * Factory for container builders.
+ *
+ * @param <T> the type of the container
+ */
+public interface ContainerFactory<T> {
+
+    /**
+     * Whether the corresponding container builder accepts objects of the given type.
+     *
+     * @param type the object type
+     * @return {@code true}, if accepts, otherwise {@code false}
+     */
+    boolean accepts(Class<?> type);
+
+    /**
+     * Builds a new container builder for objects of the given type.
+     *
+     * @param type the object type
+     * @return the container builder
+     */
+    ContainerBuilder<T> newContainerBuilderFor(Class<?> type);
```

Comment:
Huh. The wildcard argument coupled with the ContainerBuildet<T> parameterized result is a nice compromise on type safety.